circuits/circom/prover/index.ts

```diff
@@ -0,0 +1,49 @@
+import http from "http";
+import { ProofRequest, stringifyWithBigInts } from "./interface";
+import { buildPoseidon } from 'circomlibjs'
+import {
+    bigintToArray,
+    MerkleTree,
+    uint8ArrayToBigint,
+  } from '../test/helpers'
+
+const myServer = http.createServer((req, res) => {
+    // Read the data from the request
+    let data = "";
+
+    req.on("data", (chunk) => {
+        data += chunk.toString();
+    });
+
+    // When the request is done
+    req.on("end", async () => {
+        let request = ProofRequest.fromJSON(data);
+        const poseidon = await buildPoseidon()
```

Comment:
does `poseidon` need to be re built on every request?
Can we build it once and store/cache it in memory?

Reply:
No it doesn't, but it does have to run in an async function so we can await it. I really don't understand why there needs to be an asynchronous function to build poseidon, but apparently there is.

Perhaps I could call `buildPoseidon` at the top level, and just await it inside the request?

Reply:
Fixed.